pkg/odo/cli/service/create.go

```diff
@@ -425,3 +473,27 @@ func getAlmExample(almExamples []map[string]interface{}, crd, operator string) (
 	}
 	return nil, errors.Errorf("Could not find example yaml definition for %q service in %q operator's definition.\nPlease provide a file containing yaml specification to start the service from operator\n", crd, operator)
 }
+
+func doesCRExist(kind string, csvs *olmv1alpha1.ClusterServiceVersionList) (olmv1alpha1.ClusterServiceVersion, error) {
+	for _, csv := range csvs.Items {
+		for _, operatorCR := range csv.Spec.CustomResourceDefinitions.Owned {
+			if kind == operatorCR.Kind {
+				return csv, nil
+			}
+		}
+	}
+	return olmv1alpha1.ClusterServiceVersion{}, errors.New("Could not find the requested cluster resource")
+
+}
+
+func serviceNameFromCRD(crd map[string]interface{}, serviceName string) string {
+	metadata, ok := crd["metadata"].(map[string]interface{})
+	if !ok {
```

Comment:
why are we returning the serviceName if the casting fails? please add some comments here

Reply:
Using `serviceName` is probably a bad idea. Because while running this locally, I found that `serviceName` is empty as we don't pass any arguments to the command. Besides that, an operator's going to fail the creation of service if it doesn't find `name` field in `metadata`. I guess we can add a check to make sure that `metadata`->`name` exists.

This might sound like redoing what operator is already checking for us. But having the info about name _might_ come handy when listing the services in #2593.

Reply:
Here's the error if there's no name in the yaml:
```sh
$ cat /tmp/etcd.yaml
apiVersion: etcd.database.coreos.com/v1beta2
kind: EtcdCluster
spec:
  size: 3
  version: 3.2.13

$ odo service create --from-file /tmp/etcd.yaml
Deploying service of type: EtcdCluster
 ✗  Deploying service [1ms]
 ✗  EtcdCluster.etcd.database.coreos.com "" is invalid: metadata.name: Required value: name or generateName is required

```

Reply:
> Using `serviceName` is probably a bad idea. Because while running this locally, I found that `serviceName` is empty as we don't pass any arguments to the command.

`odo service create` normally picks up the service name from CLI or names it with a default: https://github.com/openshift/odo/blob/c15e2d5ebd90fc55b623a9a74a83baba7739c09b/pkg/odo/cli/service/create.go#L151-L153

But in this case, we're not going that far if `--from-file` is found in the arguments.

> Besides that, an operator's going to fail the creation of service if it doesn't find name field in metadata. I guess we can add a check to make sure that metadata->name exists.

Here's the error if there's no name in the yaml:
```sh
$ cat /tmp/etcd.yaml
apiVersion: etcd.database.coreos.com/v1beta2
kind: EtcdCluster
spec:
  size: 3
  version: 3.2.13

$ odo service create --from-file /tmp/etcd.yaml
Deploying service of type: EtcdCluster
 ✗  Deploying service [1ms]
 ✗  EtcdCluster.etcd.database.coreos.com "" is invalid: metadata.name: Required value: name or generateName is required

```

Reply:
OK